### PR TITLE
Material Builder

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Adds a material builder. Provides implementations for protected methods, a few methods not found in the original class, and a private method hook.

### CODE REVIEW

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockMaterial.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	@Override
+	public FabricBlockMaterial setCanBeBrokenInAdventureMode() {
+		super.setCanBeBrokenInAdventureMode();
+		return this;
+	}
+
+	/**
+	 * Subject to changes.
```
Comment (shedaniel) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockMaterial.java:116]: why? at least explain why in the doc

File: fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/ObjectBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+package net.fabricmc.fabric.test.object.builder;
+
+import net.minecraft.block.Material;
+import net.minecraft.block.MaterialColor;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.object.builder.v1.block.FabricBlockMaterial;
+
+public class ObjectBuilderTest implements ModInitializer {
+	public static final Material TEST_MATERIAL = new FabricBlockMaterial(MaterialColor.WHITE)
```
Comment (shedaniel) [inline fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/ObjectBuilderTest.java:26]: you gonna use the material to test the material

Comment (BoogieMonster1O1) [reply fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/ObjectBuilderTest.java:26]: read commit 6846e3c
i'll do it though

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockMaterial.java (hunk 1 above)
Comment (BoogieMonster1O1) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockMaterial.java:116]: might be replaced with a mixin or an access widener.
not really a problem for users

File: fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/ObjectBuilderTest.java (hunk 2 above)
Comment (shedaniel) [reply fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/ObjectBuilderTest.java:26]: there is no point making the test mod if you are not testing if it works.

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockMaterial.java (hunk 1 above)
Comment (shedaniel) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockMaterial.java:116]: huh?

Comment (BoogieMonster1O1) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/FabricBlockMaterial.java]: removed

File: fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/ObjectBuilderTest.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/ObjectBuilderTest.java:29]: fine

**Pull request merged**